### PR TITLE
Improve doc bit by bit (#90)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
- fix a typo in State
- simplify builder default adapter

### CODE REVIEW

File: gui/doc/layout.md [hunk 1]
```diff
@@ -1,17 +1,146 @@
...
+            ...
+        |>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:layout columns="50px 1fr 50px">
+            <taipy:part>
+                <taipy.text>{1st column content}</taipy.text>
```
Comment (dinhlongviolin1) [inline gui/doc/layout.md]: correct me if I am wrong but this should be taipy:text right

File: gui/doc/input.md [hunk 2]
```diff
@@ -2,7 +2,7 @@ A control that displays some text that can potentially be edited.
 
 ## Usage
 
-### Simple
+### Allows user input
```
Comment (FabienLelaquais) [inline gui/doc/input.md]: "Allows" -> "Get"

File: gui/doc/layout.md (hunk 1 above)
Comment (FredLL-Avaiga) [reply gui/doc/layout.md]: well spotted Long, Tx

File: gui/doc/layout.md [hunk 3]
```diff
@@ -1,17 +1,146 @@
...
+### Specifying gap
+
+The _gap_ between adjacent cells is set by default to 0.5rem and can be specified.
+
+!!! example "Page content"
+
+    === "Markdown"
+
+        ```
+        <||layout|gap=20px|
```
Comment (FabienLelaquais) [inline gui/doc/layout.md]: two || here. Is copy/paste a clue?

File: gui/doc/layout.md [hunk 4]
```diff
@@ -1,17 +1,146 @@
...
+### Default layout
+
+The default layout contains 2 columns in desktop mode and 1 column in mobile mode.
+
+!!! example "Page content"
+
+    === "Markdown"
+
+        ```
+        <||layout|
```
Comment (FabienLelaquais) [inline gui/doc/layout.md]: Two || here.

File: gui/doc/layout.md [hunk 5]
```diff
@@ -1,17 +1,146 @@
...
+
+        |>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:layout>
+
+            <taipy:text>{some content}</taipy:text>>
```
Comment (dinhlongviolin1) [inline gui/doc/layout.md]: why is there 2 >> in this line

File: gui/doc/layout.md [hunk 6]
```diff
@@ -1,17 +1,146 @@
...
+        <taipy:layout gap="20px">
+            ...
+            <taipy:text>{some content}</taipy:text>>
+            ...
+        </taipy:layout>
+        ```
+
+### Layout with a central "greedy" column
+
+The _columns_ property follows the [https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-columns](css standard).
```
Comment (FabienLelaquais) [inline gui/doc/layout.md]: That's what you said in the preambule.
May be "You can use the _fr_ CSS unit so that..."

File: gui/doc/layout.md (hunk 5 above)
Comment (FredLL-Avaiga) [reply gui/doc/layout.md]: typo

File: gui/doc/layout.md [hunk 7]
```diff
@@ -1,17 +1,146 @@
...
+                <taipy.text>{3rd column content}</taipy.text>
+            </taipy:part>
+            <taipy:part>
+                <taipy.text>{1st column and second row content}</taipy.text>
+            </taipy:part>
+            ...
+        </taipy:layout>
+        ```
+
+### Different layout for desktop and mobile view
```
Comment (FabienLelaquais) [inline gui/doc/layout.md]: view -> devices

**Review by dinhlongviolin1 — APPROVED** [review]
Everything looks good to me now!

File: gui/doc/layout.md [hunk 8]
```diff
@@ -1,17 +1,146 @@
...
+
+The _columns[mobile]_ property allows to specify a different layout when a mobile device is detected.
```
Comment (FabienLelaquais) [inline gui/doc/layout.md]: "when a mobile device is detected."
->
"when running on a mobile device."

File: gui/doc/layout.md [hunk 9]
```diff
@@ -1,17 +1,146 @@
...
+
+!!! example "Page content"
+
+    === "Markdown"
+
+        ```
+        <||layout|columns=50px 1fr 50px|columns[mobile]=1 1|
```
Comment (FabienLelaquais) [inline gui/doc/layout.md]: Two ||

File: gui/doc/menu.md [hunk 10]
```diff
@@ -1,13 +1,81 @@
...
 
 This control is represented by a unique left-anchor and foldable vertical menu.
 
 ## Usage
 
-### Simple
+### Defining a simple static menu
 
-<code><|menu|lov={["menu 1", "menu 2"]}></code>
+The property _on_action_ default value is "on_menu_action"; a function with that name will be called if present.
```
Comment (FabienLelaquais) [inline gui/doc/menu.md]: That should appear in the doc for _on_action_.
May be you need a (common) piece of Python code to show what that means in real life.

```py
def my_menu_action(state, ...):
  ...
```

and the markdown/html snippet right after

File: gui/doc/menu.md [hunk 11]
```diff
@@ -1,13 +1,81 @@
...
+  
+    === "HTML"
+
+        ```html
+        <taipy:menu lov="menu 1;menu 2" label="Menu title" width="15vw" width[mobile]="80vw"></taipy:menu>
+        ```
+
+### Menu icons
+
+As for every control that deals with lov, the menu can display Icon with label for each option.
```
Comment (FabienLelaquais) [inline gui/doc/menu.md]: "the menu can display Icon with label for each option"
->
"each menu option can display an image (see `Icon^`) and/or some text."

File: gui/doc/menu.md [hunk 12]
```diff
@@ -1,13 +1,81 @@
...
+
+!!! example "Page content"
+
+    === "Markdown"
+
+        ```
+        <|menu|lov={[("id1", Icon("/images/fred.png", "Menu option 1")), ("id2", "Menu option 2")]}|>
```
Comment (FabienLelaquais) [inline gui/doc/menu.md]: fred -> icon?

File: gui/doc/menu.md [hunk 13]
```diff
@@ -1,13 +1,81 @@
...
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:menu>{[("id1", Icon("/images/fred.png", "Menu option 1")), ("id2", "Menu option 2")]}</taipy:menu>
```
Comment (FabienLelaquais) [inline gui/doc/menu.md]: fred -> icon?

File: gui/doc/menu.md [hunk 14]
```diff
@@ -1,13 +1,81 @@
...
 This control is represented by a unique left-anchor and foldable vertical menu.
 
 ## Usage
 
-### Simple
+### Defining a simple static menu
 
-<code><|menu|lov={["menu 1", "menu 2"]}></code>
+The property _on_action_ default value is "on_menu_action"; a function with that name will be called if present.
+The parameter payload["args"] holds a list that contains the id of the selected menu.
```
Comment (FabienLelaquais) [inline gui/doc/menu.md]: And this should be in the doc as well.
Now, let's use it!

**Review by FabienLelaquais — APPROVED** [review]
Minor yet boring stuff

**Pull request merged**